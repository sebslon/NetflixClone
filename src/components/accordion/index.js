import React, { createContext, useState, useContext } from "react";

import {
  Body,
  Container,
  Header,
  Item,
  Inner,
  Title,
} from "./styles/accordion";

const ToggleContext = createContext();

export default function Accordion({ children, ...otherProps }) {
  return (
    <Container {...otherProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function AccordionTitle({ children, ...otherProps }) {
  return <Title {...otherProps}>{children}</Title>;
};

Accordion.Item = function AccordionItem({ children, ...otherProps }) {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...otherProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader({ children, ...otherProps }) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header
      onClick={() => setToggleShow((prevState) => !prevState)}
      {...otherProps}
    >
      {children}
      {toggleShow ? (
        <img src="images/icons/close-slim.png" alt="close" />
      ) : (
        <img src="images/icons/add.png" alt="open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...otherProps }) {
  const { toggleShow } = useContext(ToggleContext);

  return toggleShow ? <Body {...otherProps}>{children}</Body> : null;
};
