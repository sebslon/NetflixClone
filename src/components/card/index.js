import { createContext, useContext, useState } from "react";

import {
  Container,
  Content,
  Entities,
  Feature,
  FeatureClose,
  FeatureTitle,
  FeatureText,
  Group,
  Item,
  Image,
  Maturity,
  Meta,
  SubTitle,
  Title,
  Text,
} from "./styles/card";

export const FeatureContext = createContext();

export default function Card({ children, ...otherProps }) {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...otherProps}>{children}</Container>
    </FeatureContext.Provider>
  );
}

Card.Group = function CardGroup({ children, ...otherProps }) {
  return <Group {...otherProps}>{children}</Group>;
};

Card.Title = function CardTitle({ children, ...otherProps }) {
  return <Title {...otherProps}>{children}</Title>;
};

Card.SubTitle = function CardSubTitle({ children, ...otherProps }) {
  return <SubTitle {...otherProps}>{children}</SubTitle>;
};

Card.Text = function CardText({ children, ...otherProps }) {
  return <Text {...otherProps}>{children}</Text>;
};

Card.Entities = function CardEntity({ children, ...otherProps }) {
  return <Entities {...otherProps}>{children}</Entities>;
};

Card.Meta = function CardMeta({ children, ...otherProps }) {
  return <Meta {...otherProps}>{children}</Meta>;
};

Card.Feature = function CardFeature({ children, category, ...otherProps }) {
  const { showFeature, itemFeature, setShowFeature } = useContext(FeatureContext);

  return showFeature ? (
    <Feature {...otherProps}
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature(false)}>
          <img src="/images/icons/close.png" alt="close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity > 12 ? "PG" : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() + itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null;
};

Card.Item = function CardItem({ item, children, ...otherProps }) {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext);

  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...otherProps}
    >
      {children}
    </Item>
  );
};

Card.Image = function CardImage({ ...otherProps }) {
  return <Image {...otherProps} />;
};
