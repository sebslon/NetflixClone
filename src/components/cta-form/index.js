import { Break, Container, Input, Button, Text } from './styles/cta-form';

export default function CtaForm({ children, ...otherProps }) {
  return (
    <Container {...otherProps}>{children}</Container>
  )
}

CtaForm.Input = function CtaFormInput({ ...otherProps }) {
  return <Input {...otherProps} />
}

CtaForm.Button = function CtaFormButton({ children, ...otherProps }) {
  return (
    <Button {...otherProps}>
      {children} <img src="images/icons/chevron-right.png" alt="Try now" />
    </Button>
  )
};

CtaForm.Text = function CtaFormText({ children, ...otherProps }) {
  return <Text {...otherProps}>{children}</Text>
}

CtaForm.Break = function CtaFormBreak({...otherProps}) {
  return <Break {...otherProps} />;
}