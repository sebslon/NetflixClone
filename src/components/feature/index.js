import { Container, Title, SubTitle } from './styles/feature';

export default function Feature({children, ...otherProps}) {
  return <Container {...otherProps}>{children}</Container>
};

Feature.Title = function FeatureTitle({children, ...otherProps}) {
  return <Title {...otherProps}>{children}</Title>
}

Feature.SubTitle = function FeatureSubTitle({children, ...otherProps}) {
  return <SubTitle {...otherProps}>{children}</SubTitle>
}