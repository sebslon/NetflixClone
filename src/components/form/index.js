import { Base, Container, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form';

export default function Form({ children, ...otherProps }) {
  return <Container {...otherProps}>{children}</Container>
}

Form.Base = function FormBase({ children, ...otherProps }) {
  return <Base {...otherProps}>{children}</Base>
}

Form.Error = function FormError({ children, ...otherProps }) {
  return <Error {...otherProps}>{children}</Error>
}

Form.Title = function FormTitle({ children, ...otherProps }) {
  return <Title {...otherProps}>{children}</Title>
}

Form.Text = function FormText({ children, ...otherProps }) {
  return <Text {...otherProps}>{children}</Text>
}

Form.TextSmall = function FormTextSmall({ children, ...otherProps }) {
  return <TextSmall {...otherProps}>{children}</TextSmall>
}

Form.Link = function FormLink({ children, ...otherProps }) {
  return <Link {...otherProps}>{children}</Link>
}

Form.Input = function FormInput({ children, ...otherProps }) {
  return <Input {...otherProps}>{children}</Input>
}

Form.Submit = function FormSubmit({ children, ...otherProps }) {
  return <Submit {...otherProps}>{children}</Submit>
}