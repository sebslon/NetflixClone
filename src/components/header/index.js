import { Link } from 'react-router-dom';

import { Background, Container, ButtonLink, Logo } from './styles/header';

export default function Header({ background = true, children, ...otherProps }) {
  return background ? <Background {...otherProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...otherProps }) {
  return <Container {...otherProps}>{children}</Container>;
}

Header.ButtonLink = function HeaderButtonLink({ children, ...otherProps }) {
  return <ButtonLink {...otherProps}>{children}</ButtonLink>;
}

Header.Logo = function HeaderLogo({ to, ...otherProps }) {
  return (
    <Link to={to}>
      <Logo {...otherProps} />
    </Link>
  )
}
