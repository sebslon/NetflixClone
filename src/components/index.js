export { default as Accordion } from './accordion'
export { default as CtaForm } from './cta-form'
export { default as Footer } from './footer';
export { default as Jumbotron } from './jumbotron';
