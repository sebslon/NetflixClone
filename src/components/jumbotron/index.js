import React from 'react';

import { Container, Inner, Item, Pane, Title, SubTitle, Image } from './styles/jumbotron';

export default function Jumbotron({ children, direction = 'row', ...otherProps }) {
  return (
    <Item>
      <Inner direction={direction}>
        {children}
      </Inner>
    </Item>
  )
}

Jumbotron.Container = function JumbotronContainer({ children, ...otherProps }) {
  return <Container {...otherProps}>{children}</Container>
}

Jumbotron.Pane = function JumbotronPane({ children, ...otherProps }) {
  return <Pane {...otherProps}>{children}</Pane>
}

Jumbotron.Title = function JumbotronTitle({ children, ...otherProps }) {
  return <Title {...otherProps}>{children}</Title>
}

Jumbotron.SubTitle = function JumbotronSubTitle({ children, ...otherProps }) {
  return <SubTitle {...otherProps}>{children}</SubTitle>
}

Jumbotron.Image = function JumbotronImage({ ...otherProps }) {
  return <Image {...otherProps} />
}