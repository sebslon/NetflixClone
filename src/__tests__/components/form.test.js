import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import { Form } from "../../components";

describe("<Form />", () => {
  it("Renders <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Router>
        <Form>
          <Form.Title>Sign In Now</Form.Title>
          <Form.Base>
            <Form.Input placeholder="Email address" onChange={() => {}} />
            <Form.Input
              type="password"
              placeholder="Password"
              onChange={() => {}}
            />
            <Form.Submit type="submit" disabled>
              Sign In
            </Form.Submit>
          </Form.Base>

          <Form.Text>
            New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
          </Form.Text>

          <Form.TextSmall>
            This page is protected by Google reCAPTCHA to ensure you're not a
            bot. Learn more.
          </Form.TextSmall>
        </Form>
      </Router>
    );

    expect(
      getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();

    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByPlaceholderText("Email address")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the <Form /> with an error", () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Your email address is already in use.</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );

    expect(getByText("Your email address is already in use.")).toBeTruthy();
  });
});
