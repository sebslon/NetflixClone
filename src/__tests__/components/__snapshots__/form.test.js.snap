// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Renders <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-7952rh-0 SEPfC"
>
  <h1
    class="form__Title-sc-7952rh-3 fCKYwV"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7952rh-2 klyPny"
  >
    <input
      class="form__Input-sc-7952rh-7 gUryur"
      placeholder="Email address"
    />
    <input
      class="form__Input-sc-7952rh-7 gUryur"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 kWcsU"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7952rh-4 fxUZEo"
  >
    New to Netflix? 
    <a
      class="form__Link-sc-7952rh-6 fZbchy"
      href="/signup"
    >
      Sign up now.
    </a>
  </p>
  <p
    class="form__TextSmall-sc-7952rh-5 iEgvHc"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
